src/ilasm/CMakeLists.txt

```diff
@@ -19,31 +20,55 @@ set(ILASM_SOURCES
   assembler.cpp
   prebuilt/asmparse.c
 )
-if(WIN32)
-  set_source_files_properties(prebuilt/asmparse.c PROPERTIES COMPILE_FLAGS "/TP")
-endif(WIN32)
+set_source_files_properties( prebuilt/asmparse.c PROPERTIES LANGUAGE CXX )
+
+if(CLR_CMAKE_PLATFORM_UNIX)
+  add_compile_options(-x c++)
```

Comment:
I wonder - is the `-x C++` still needed when you've set the asmparse.c language to CXX?